int64 idKF
int64 idMP